opencti-platform/opencti-graphql/src/modules/case/case-incident/case-incident-domain.ts

```diff
@@ -0,0 +1,52 @@
+import type { AuthContext, AuthUser } from '../../../types/user';
+import { createEntity } from '../../../database/middleware';
+import type { EntityOptions } from '../../../database/middleware-loader';
+import { internalLoadById, listEntitiesPaginated, storeLoadById } from '../../../database/middleware-loader';
+import { BUS_TOPICS } from '../../../config/conf';
+import { ABSTRACT_STIX_DOMAIN_OBJECT, buildRefRelationKey } from '../../../schema/general';
+import { notify } from '../../../database/redis';
+import { now } from '../../../utils/format';
+import { addIndividual } from '../../../domain/individual';
+import { userSessionRefresh } from '../../../domain/user';
+import { isEmptyField } from '../../../database/utils';
+import type { BasicStoreEntityCaseIncident } from './case-incident-types';
+import { ENTITY_TYPE_CONTAINER_CASE_INCIDENT } from './case-incident-types';
+import type { DomainFindById } from '../../../domain/domainTypes';
+import type { CaseIncidentAddInput } from '../../../generated/graphql';
+import { isStixId } from '../../../schema/schemaUtils';
+import { RELATION_OBJECT } from '../../../schema/stixMetaRelationship';
+
+export const findById: DomainFindById<BasicStoreEntityCaseIncident> = (context: AuthContext, user: AuthUser, caseIncidentId: string) => {
+  return storeLoadById(context, user, caseIncidentId, ENTITY_TYPE_CONTAINER_CASE_INCIDENT);
+};
+
+export const findAll = (context: AuthContext, user: AuthUser, opts: EntityOptions<BasicStoreEntityCaseIncident>) => {
+  return listEntitiesPaginated<BasicStoreEntityCaseIncident>(context, user, [ENTITY_TYPE_CONTAINER_CASE_INCIDENT], opts);
+};
+
+export const addCaseIncident = async (context: AuthContext, user: AuthUser, caseIncidentAdd: CaseIncidentAddInput) => {
+  let caseToCreate = caseIncidentAdd.created ? caseIncidentAdd : { ...caseIncidentAdd, created: now() };
+  if (isEmptyField(caseIncidentAdd.createdBy)) {
+    let individualId = user.individual_id;
+    if (individualId === undefined) {
+      const individual = await addIndividual(context, user, { name: user.name, contact_information: user.user_email });
```

Comment:
You can use the new method in user.js : userAddIndividual

Reply:
Done ✅